vault/expiration.go

```diff
@@ -453,6 +446,41 @@ func (c *Core) stopExpiration() error {
 	return nil
 }
 
+func (m *ExpirationManager) leaseView(ns *namespace.Namespace) BarrierView {
+	if ns.ID == namespace.RootNamespaceID {
+		return m.core.systemBarrierView.SubView(expirationSubPath + leaseViewPrefix)
+	}
+	return m.core.namespaceMountEntryView(ns, systemBarrierPrefix+expirationSubPath+leaseViewPrefix)
+}
+
+func (m *ExpirationManager) tokenIndexView(ns *namespace.Namespace) BarrierView {
+	if ns.ID == namespace.RootNamespaceID {
+		return m.core.systemBarrierView.SubView(expirationSubPath + tokenViewPrefix)
+	}
+	return m.core.namespaceMountEntryView(ns, systemBarrierPrefix+expirationSubPath+tokenViewPrefix)
+}
+
+func (m *ExpirationManager) collectLeases() (map[*namespace.Namespace][]string, int, error) {
```

Comment:
@voigt Do you want to file a follow-up issue to make this work with a callback? Then after this merges and #1102 merges, we can update this to not load all leases into memory :-) 

That will need to be a separate RFC to improve scalability of this particular subsystem.